dvc/repo/plots/__init__.py

```diff
@@ -273,11 +243,27 @@ def _collect_plots(
         rev=rev,
         recursive=recursive,
     )
+
     result = {plot.path_info: _plot_props(plot) for plot in plots}
     result.update({path_info: {} for path_info in path_infos})
     return result
 
 
+@error_handler
+def parse(fs, path, props=None, rev_props=None, **kwargs):
+    props = props or {}
+    rev_props = rev_props or {}
+    _, extension = os.path.splitext(path)
+    if extension in (".tsv", ".csv"):
+        header = {**rev_props, **props}.get("header", True)
+        if extension == ".csv":
+            return _load_sv(path=path, fs=fs, delimiter=",", header=header)
+        return _load_sv(path=path, fs=fs, delimiter="\t", header=header)
+    if extension in LOADERS or extension in (".yml", ".yaml"):
+        return LOADERS[extension](path=path, fs=fs)
+    raise PlotMetricTypeError(path)
```

Comment:
Moved reading plots files here. (Before it was in to_datapoints). We are moving parsing here to catch potential parsing error here, and not in render level. We could probably get rid of to_datapoints method at this point, but I am leaving it as is to not complicate this PR further. Logic behind `to_datapoints` is strictly related to file structure (eg csv is list of datapoints while json can be a dict that needs to be explored for data) so it still makes sense to relate data transformation to file extension (which is what `to_datapoints` do now), but it seems to me that we will need to change the approach how we deal with plots data.

Previously it was:
1. read datafiles from revisions
2. parse, transform and render (in render method)

Now it is:
1. read and parse datafiles from revsions
2. transform and render

How it will need to be, when we start introducing images support:
1. read and parse (if applicable)
2. transform (if applicable)
3. render